### DISCUSSION TITLE
How does work specifying map of annotations?

### DISCUSSION THREAD

**Discussion by dSalieri** [mobxjs/mobx#3740] [General] [answered]
I am trying to sort out how annotations work in `makeObservable` function.

For example we have the next:
```JavaScript
import { makeObservable, observable, action } from '/mobx.js';
makeObservable({
  counter: 0, 
  inc() {
    this.counter++;
  }
}, {
  counter: observable,
  inc: action
})
```

`observable` and `action` are functions and they must be called as I expected. But they have not been invoked. How does actually work `makeObservable` if it doesn't use functions from annotations map?

**Comment by urugator** [comment, marked as answer]
When you use these functions as annotations, you should think of them as symbols, like so:
```js
if (annotation === observable) {
  // some logic that defines observable prop
}
```
In the actual implementation, there are extra props defined on these functions that are being used by `makeObservable`, but that's an implementation detail. You can't implement your own annotation, there is no public `Annotation` interface.
